Remove special cross compilation handling of everestpy and update pybind11

hikinggrass opened the pull request:
This fixes an issue with pybind11 not finding the appropriate python environment during cross compilation

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

hikinggrass committed:
Move cmake cross compilation handling of everestpy into toplevel cmake file

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

a-w50 commented on CMakeLists.txt:
```diff
@@ -29,6 +29,17 @@ find_package(Boost
+    set(PYTHON_MODULE_EXTENSION ".so" CACHE INTERNAL "")
+
+    set(PYTHONLIBS_FOUND TRUE CACHE INTERNAL "")
+endif()
```
are these EVEREST_CROSS variables used anywhere else? If not, I would drop them and set the PYTHON_* variables directly, instead of setting them with the EVEREST_CROSS variables.  Furthermore, does yocto build with this hardcoded setup? I doubt it, therefore I would put it after `if(NOT DISABLE_EDM)`, but before `evc_setup_edm()`.  There is also https://cmake.org/cmake/help/latest/module/FindPython3.html which could be possibly used to find the correct files right away

hikinggrass replied:
At the moment these EVEREST_CROSS_* variables are used in the kirkstone-next branch of meta-everest. I would also prefer to either set the PYTHON_* variables directly or investigate the use of the FindPython3 module to do this automatically, this was just the least intrusive and quickest way of fixing cross compiling earlier

a-w50 committed:
Cleanup

Signed-off-by: aw <aw@pionix.de>

a-w50 committed:
Fix

Signed-off-by: aw <aw@pionix.de>

hikinggrass committed:
Remove CMAKE_CROSSOMPILING handling for pybind11
The toolchain should either set the following variables automatically or they should be provided via the command line
PYTHON_INCLUDE_DIRS
PYTHON_LIBRARIES
PYTHON_MODULE_EXTENSION
PYTHONLIBS_FOUND

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

hikinggrass committed:
Upgrade pybind11 dependency version to v2.10.3

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>
